db/commitlog/commitlog.cc

```diff
@@ -1683,6 +1710,9 @@ void db::commitlog::segment_manager::on_timer() {
         if (cfg.mode != sync_mode::BATCH) {
             sync();
         }
+
+        _sync_semaphore.signal(_sync_semaphore.waiters());
+
```

Comment:
What if different waiters are waiting for different times?

Reply:
They will be woken up at the earliest time someone forces flush to happen. My assumption is that the wait time is a max latency we are willing to add, but if things move along before that we are happy.

Reply:
Yes, makes sense.